### PR TITLE
Rewrite

### PR DESCRIPTION
**Pull request by UziTech**
I rewrote most of the package how I would create it:

- Took out the toggle command (users can just disable the package if they don't want brackets colored)
- Changed the classes to be more verbose and avoid collisions with other packages
- Changed the setting format to match other bracket packages
- use a WeakMap to save editor objects instead of on the editor object
- Use atom-jasmine3-test-runner for tests and added a few working tests

There are a lot of breaking changes so users might be annoyed at first but I think in the long run it would be better.

### CODE REVIEW

File: lib/bracket-colorizer.js [hunk 1]
```diff
@@ -1,95 +1,78 @@
...
   subscriptions: null,
+  editors: new WeakMap(),
+  brackets: null,
 
   config: {
     brackets: {
       title: 'Brackets to match',
-      description: "The format should be, '{symbol_start};{symbol_end}'",
-      default: ['{;}', '(;)', '[;]'],
+      description: "The format should be: {}, (), []",
```
Comment (vn-ki) [inline lib/bracket-colorizer.js]: Maybe something more descriptive? The user may want to include symbols other than `{}, (), []`.

File: lib/bracket-colorizer.js [hunk 2]
```diff
@@ -1,95 +1,78 @@
...
-    this.on = false;
-
+  activate() {
     this.subscriptions = new CompositeDisposable();
-    this.subscriptions.add(atom.commands.add('atom-workspace', {
-      'bracket-colorizer:toggle': () => this.toggle()
+    this.editors = new WeakMap();
+
+    this.subscriptions.add(atom.workspace.onDidStopChangingActivePaneItem(() => this.processEditor()));
+    this.subscriptions.add(atom.config.observe('bracket-colorizer.brackets', (value) => {
```
Comment (vn-ki) [inline lib/bracket-colorizer.js]: Isn't `this.brackets = atom.config.get('bracket-colorizer.bracket')` enough? I think atom reloads the package after config change.

**Comment by vn-ki** [comment]
Thanks for this. The classes make more sense this way. If we have to add more functionality in the future this would help too.

Is there any specific reason to remove toggle? The presence seemed of it seemed useful to me. Also, keeping it would ensure the users won't find the change too disturbing.

File: lib/bracket-colorizer.js (hunk 2 above)
Comment (UziTech) [reply lib/bracket-colorizer.js]: Nope, Atom does not reload the package.

**Comment by UziTech** [comment]
The toggle command seems redundant since you can toggle the package by enabling/disabling it in the settings and if someone wants it on all the time, (which I would guess would be most users) they have to toggle it every time Atom starts.

File: lib/bracket-colorizer.js (hunk 1 above)
Comment (UziTech) [reply lib/bracket-colorizer.js]: I changed to to be more like [bracket-matcher](https://github.com/atom/bracket-matcher/blob/master/package.json#L20)

**Comment by vn-ki** [comment]
Merged!

File: lib/bracket-colorizer.js (hunk 2 above)
Comment (vn-ki) [reply lib/bracket-colorizer.js]: > this.brackets = atom.config.get('bracket-colorizer.bracket')

This seems to work for me.

Comment (UziTech) [reply lib/bracket-colorizer.js]: `atom.config.get` only gets the value of the setting when it is called.

`atom.config.observe` is used to observe changes to a setting.

If we use `this.brackets = atom.config.get('bracket-colorizer.bracket')` then `this.brackets` won't be updated until the next time we call that line

Comment (vn-ki) [reply lib/bracket-colorizer.js]: That's weird! I changed `observe` to `get`, reloaded and then changed the settings. It changed.

**Pull request merged**